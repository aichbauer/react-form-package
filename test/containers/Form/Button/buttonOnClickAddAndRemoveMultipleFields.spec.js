import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with button | add and remove fields | click button', () => {
  const onClickMockRemoveField = jest.fn();
  const onClickMockAddField = jest.fn();

  const streets = [
    {
      id: 'street-0',
    },
  ];

  const myComponent = (
    <Form>
      <>
        {streets.map((street) => (
          <div
            key={street.id}
          >
            <Field
              id={street.id}
              type="text"
              required
            />
            <Field
              id={`housenumber-${street.id.split('-')[1]}`}
              type="number"
              required
            />
            <Button
              id="removeField"
              rfpRole="removeField"
              type="button"
              fieldId={[
                street.id,
                `housenumber-${street.id.split('-')[1]}`,
              ]}
              onClick={onClickMockRemoveField}
            >
              Remove Company
            </Button>
          </div>
        ))}
      </>
      <div>
        <Button
          id="addField"
          rfpRole="addField"
          type="button"
          field={[
            {
              id: 'street-1',
              type: 'text',
              required: true,
            },
            {
              id: 'housnumber-1',
              type: 'number',
              required: true,
            },
          ]}
          onClick={onClickMockAddField}
        >
          Add Company
        </Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(Object.keys(state.data).length).toBe(2);

  wrapper.find('#addField').at(0).simulate('click');
  wrapper.update();

  state = wrapper.state();
  expect(Object.keys(state.data).length).toBe(4);

  wrapper.find('#removeField').at(0).simulate('click');
  wrapper.update();

  state = wrapper.state();
  expect(Object.keys(state.data).length).toBe(2);

  expect(onClickMockRemoveField).toHaveBeenCalled();
  expect(onClickMockAddField).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
