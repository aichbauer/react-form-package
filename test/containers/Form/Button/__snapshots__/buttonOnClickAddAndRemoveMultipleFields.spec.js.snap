// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with button | add and remove fields | click button 1`] = `
<form>
  <div>
    <input
      className="rfp-input"
      id="street-0"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="text"
      valid={false}
      value=""
    />
    <input
      className="rfp-input"
      id="housenumber-0"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="number"
      valid={false}
      value=""
    />
    <button
      className="rfp-button"
      disabled={false}
      fieldId={
        Array [
          "street-0",
          "housenumber-0",
        ]
      }
      id="removeField"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      rfpRole="removeField"
      type="button"
    >
      Remove Company
    </button>
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      field={
        Array [
          Object {
            "id": "street-1",
            "required": true,
            "type": "text",
          },
          Object {
            "id": "housnumber-1",
            "required": true,
            "type": "number",
          },
        ]
      }
      id="addField"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      rfpRole="addField"
      type="button"
    >
      Add Company
    </button>
  </div>
</form>
`;
